src/routes/safe/components/Transactions/TxList/hooks/useTransactionType.ts

```diff
@@ -1,22 +1,30 @@
 import { useEffect, useState } from 'react'
 import { useSelector } from 'react-redux'
 
-import { Transaction } from 'src/logic/safe/store/models/types/gateway.d'
+import { Transaction, Custom } from 'src/logic/safe/store/models/types/gateway.d'
 import { safeParamAddressFromStateSelector } from 'src/logic/safe/store/selectors'
 import CustomTxIcon from 'src/routes/safe/components/Transactions/TxList/assets/custom.svg'
 import CircleCrossRed from 'src/routes/safe/components/Transactions/TxList/assets/circle-cross-red.svg'
 import IncomingTxIcon from 'src/routes/safe/components/Transactions/TxList/assets/incoming.svg'
 import OutgoingTxIcon from 'src/routes/safe/components/Transactions/TxList/assets/outgoing.svg'
 import SettingsTxIcon from 'src/routes/safe/components/Transactions/TxList/assets/settings.svg'
+import { getTxTo } from 'src/routes/safe/components/Transactions/TxList/utils'
+import { useKnownAddress } from './useKnownAddress'
 
 export type TxTypeProps = {
-  icon: string | null
-  text: string
+  icon: string | undefined
+  text: string | undefined
```

Comment:
Maybe mark them as optional? Or do we want to emphasize they can be passed as undefined?
```
icon?: string;
text?: string;
```

Reply:
changed